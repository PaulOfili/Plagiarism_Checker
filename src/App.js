import React from 'react';
import { Provider } from 'react-redux';
import store from './store';
import LayoutHandler from './layouts/layoutHandler';
import './styles/main.scss';

class  App extends React.Component {

  componentDidMount() {
      window.addEventListener('resize', this.resize);
  }

  componentWillUnmount() {
      window.removeEventListener('resize', this.resize);
  }

  resize = () => this.forceUpdate();


  render() {

    return (
        <Provider store={store}>
          <LayoutHandler />
        </Provider>
    );
  }
}

export default App;
