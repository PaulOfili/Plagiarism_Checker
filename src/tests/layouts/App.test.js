import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../../App';

Enzyme.configure({adapter: new Adapter() })

function setup() {
    
    const enzymeWrapper = shallow(<App />);

    return {
        enzymeWrapper
    }
}

describe('Layout', () => {
    describe('App', () => {
        xit('should match the snapshot', () => {
            const {enzymeWrapper} = setup()
            expect(enzymeWrapper.html()).toMatchSnapshot();
        })
    })
})