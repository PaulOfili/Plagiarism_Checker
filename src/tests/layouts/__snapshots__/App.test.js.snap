// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout App should match the snapshot 1`] = `"<div class=\\"toast-container\\"><button class=\\"toast-dismiss\\"> x </button></div>Welcome to the Landing Page of Corporate Affairs Commission.<div><form action=\\"test/oauth/authorize\\" style=\\"display:flex;align-items:center;justify-content:center\\" method=\\"get\\"><input type=\\"text\\" name=\\"client_id\\" value=\\"1234\\" readonly=\\"\\" hidden=\\"\\"/><input type=\\"text\\" name=\\"redirect_uri\\" value=\\"test\\" readonly=\\"\\" hidden=\\"\\"/><input type=\\"text\\" name=\\"response_type\\" value=\\"token\\" readonly=\\"\\" hidden=\\"\\"/><input type=\\"text\\" name=\\"scope\\" value=\\"profile\\" readonly=\\"\\" hidden=\\"\\"/><input type=\\"text\\" name=\\"state\\" value=\\"ck4e6j29s0000acnbhpttbu71\\" readonly=\\"\\" hidden=\\"\\"/><button type=\\"submit\\">Sign In</button></form></div>"`;
